src/components/designer/taro/actions/MintAssetModal.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect } from 'react';
+import { useAsyncCallback, useAsync } from 'react-async-hook';
+import { Checkbox, Form, InputNumber, Modal, Input, Select, Alert } from 'antd';
+import { SelectValue } from 'antd/lib/select';
+import { usePrefixedTranslation } from 'hooks';
+import { TarodNode, Status } from 'shared/types';
+import { MintAssetPayload } from 'store/models/taro';
+import { useStoreActions, useStoreState } from 'store';
+import { Network } from 'types';
+import { TARO_MIN_LND_BALANCE } from 'utils/constants';
+
+interface Props {
+  network: Network;
+}
+const MintAssetModal: React.FC<Props> = ({ network }) => {
+  const { l } = usePrefixedTranslation('cmps.designer.taro.actions.MintAssetModal');
+
+  const { visible } = useStoreState(s => s.modals.mintAsset);
+  const { selectedNode } = useStoreState(s => s.designer);
+  const { nodes: lightningNodes } = useStoreState(s => s.lightning);
+  const { nodes: taronodes } = useStoreState(s => s.taro);
+
+  const { hideMintAsset } = useStoreActions(s => s.modals);
+  const { mintAsset } = useStoreActions(s => s.taro);
+  const { getWalletBalance } = useStoreActions(s => s.lightning);
+  const { notify } = useStoreActions(s => s.app);
+
+  const [isCollectible, setIsCollectible] = useState<boolean>();
+  const [isAssetNameUnique, setIsAssetNameUnique] = useState<boolean>(true);
+
+  const [isLNDBalanceLow, setIsLNDBalanceLow] = useState<boolean>();
+  const [autoDepositFunds, setAutoDepositFunds] = useState<boolean>(false);
+  const [assetName, setAssetName] = useState('My New Asset');
+  const [form] = Form.useForm();
+
+  const thisTaroNode = network.nodes.taro.find(
+    node => node.name === selectedNode.id,
```

Comment:
My goal is to build a function, which I have in another branch, a function that gets the selected node directly from the designer to avoid passing in a node name to the modal.

Is this a bad idea?

Reply:
I think it's better to pass the node name to to the modal for now to be consistent. If you want to do a separate refactoring PR and update the behavior for all modals, I wouldn't object to it.

Reply:
I'm a fan of consistency. 
If you think in the future this idea would be of use, Ill do the refactor.